tsdb/tsi1/sql_index_exporter.go

```diff
@@ -0,0 +1,246 @@
+package tsi1
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"strings"
+	"unicode/utf8"
+
+	"github.com/influxdata/influxdb/tsdb"
+	"go.uber.org/zap"
+)
+
+// SQLIndexExporter writes out all TSI data for an index to a SQL export.
+type SQLIndexExporter struct {
+	w io.Writer
+
+	initialized bool
+
+	// Logs non-fatal warnings.
+	Logger *zap.Logger
+
+	// Write schema, if true.
+	ShowSchema bool
+}
+
+// NewSQLIndexExporter returns a new instance of SQLIndexExporter.
+func NewSQLIndexExporter(w io.Writer) *SQLIndexExporter {
+	return &SQLIndexExporter{
+		w: w,
+
+		Logger:     zap.NewNop(),
+		ShowSchema: true,
+	}
+}
+
+// Close ends the export and writes final output.
+func (e *SQLIndexExporter) Close() error {
+	return nil
+}
+
+// ExportIndex writes all blocks of the TSM file.
+func (e *SQLIndexExporter) ExportIndex(idx *Index) error {
+	if err := e.initialize(); err != nil {
+		return err
+	}
+
+	fmt.Fprintln(e.w, `BEGIN TRANSACTION;`)
+
+	// Iterate over each measurement across all partitions.
+	itr, err := idx.MeasurementIterator()
+	if err != nil {
+		return err
+	} else if itr == nil {
+		return nil
+	}
+	defer itr.Close()
+
+	for {
+		name, err := itr.Next()
+		if err != nil {
+			return err
+		} else if name == nil {
+			break
+		}
+
+		if err := e.exportMeasurement(idx, name); err != nil {
+			return err
+		}
+	}
+
+	fmt.Fprintln(e.w, "COMMIT;")
+	return nil
+}
+
+func (e *SQLIndexExporter) exportMeasurement(idx *Index, name []byte) error {
+	// Log measurements that can't be parsed into org/bucket.
+	if len(name) != 16 {
```

Comment:
I don't agree with this. I think we want to export the measurements as we find them in the index.